airflow/contrib/hooks/sqoop_hook.py

```diff
@@ -14,210 +14,282 @@
         :return: handle to subprocess
         """
-        prefixed_cmd = self._prepare_command(cmd, export=export)
-        return subprocess.Popen(prefixed_cmd, **kwargs)
-
-    def _prepare_command(self, cmd, export=False):
+        process = subprocess.Popen(cmd,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE,
+                                   **kwargs)
+        output, stderr = process.communicate()
+
+        if process.returncode != 0:
+            raise AirflowException((
+                                       "Cannot execute {} on {}. Error code is: {}"
+                                       "Output: {}, Stderr: {}"
+                                   ).format(cmd, self.conn.host,
+                                            process.returncode, output,
+                                            stderr))
+
+    def _prepare_command(self, export=False):
         if export:
-            connection_cmd = ["sqoop", "export", "--verbose"]
+            connection_cmd = ["sqoop", "export"]
         else:
-            connection_cmd = ["sqoop", "import", "--verbose"]
+            connection_cmd = ["sqoop", "import"]
 
+        if self.verbose:
+            connection_cmd += ["--verbose"]
         if self.job_tracker:
             connection_cmd += ["-jt", self.job_tracker]
         if self.conn.login:
             connection_cmd += ["--username", self.conn.login]
-        # todo: put this in a password file
         if self.conn.password:
             connection_cmd += ["--password", self.conn.password]
-        if self.lib_jars:
-            connection_cmd += ["-libjars", self.lib_jars]
+        if self.password_file:
+            connection_cmd += ["--password-file", self.password_file]
+        if self.libjars:
+            connection_cmd += ["-libjars", self.libjars]
         if self.files:
             connection_cmd += ["-files", self.files]
         if self.namenode:
             connection_cmd += ["-fs", self.namenode]
         if self.archives:
             connection_cmd += ["-archives", self.archives]
+        if self.num_mappers:
+            connection_cmd += ["--num-mappers", self.num_mappers]
+
+        for key, value in self.properties.items():
+            connection_cmd += ["-D", "{}={}".format(key, value)]
 
-        connection_cmd += ["--connect", "{}:{}/{}".format(self.conn.host, self.conn.port,
-                                                          self.conn.schema)]
-        connection_cmd += cmd
+        connection_cmd += ["--connect", "{}:{}/{}".format(
+            self.conn.host,
+            self.conn.port,
+            self.conn.schema
+        )]
 
         return connection_cmd
 
-    def _import_cmd(self, target_dir,
-                    append=False, type="text",
-                    num_mappers=None, split_by=None):
+    def _get_export_format_argument(self, file_type='text'):
```

Comment:
just to bring up the point of orc format, not sure what the best way to incorporate that is, but if one uses hcatalog to create a table that is orc-formatted then they need `--hcatalog-storage-stanza "stored as orcfile"`. I did try adding in this command as well as having `--as-textfile` and it seemed to still make it orc which is good! but would you think that option should be added here?

Reply:
This is more about the internals of sqoop itself. I'm using the sqoop operator to read a parquet file and insert it into Postgres. I would add the `--hcatalog-storage-stanza` to the `_export_cmd` function.

Reply:
Agreed it's more about Sqoop itself. And also was thinking same thing. I just added the option to pass extra commands, so we don't need to create an argument for each option available.